### PR TITLE
Unify naming of syscall handlers, add docs

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Related but not closing https://github.com/foundry-rs/starknet-foundry/issues/750

### CODE REVIEW

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 1]
```diff
@@ -62,8 +63,10 @@ impl From<&StarknetContractArtifacts> for ContractArtifacts {
     }
 }
 
-pub struct CairoHintProcessor<'a> {
-    pub blockifier_syscall_handler: SyscallHintProcessor<'a>,
+// This hint processor provides an implementation logic for functions from std_forge library.
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:66]:
```suggestion
// This hint processor provides an implementation logic for functions from snforge_std library.
```

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 2]
```diff
@@ -62,8 +63,10 @@ impl From<&StarknetContractArtifacts> for ContractArtifacts {
...
+// If cannot execute a hint it falls back to the CheatableSyscallHandler
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:67]:
```suggestion
// It uses CheatableSyscallHandler to execute starknet related hints
```
At least that's how I understand it

File: crates/cheatnet/src/execution/syscalls.rs [hunk 3]
```diff
@@ -109,6 +109,10 @@ pub fn call_contract_syscall(
     // endregion
 }
 
+// This hint processor modifies the standard syscalls implementation to react upon changes
+// introduced by cheatcodes like returns mocked data
```
Comment (piotmag769) [inline crates/cheatnet/src/execution/syscalls.rs:113]:
```suggestion
// introduced by cheatcodes that e.g. returns mocked data
```

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/src/cheatcodes_hint_processor.rs:67]: I am not sure if it will be true, I will keep in mind and update it later after I finish.

**Review by Arcticae — APPROVED** [review]

**Pull request merged**